app/models/course/assessment/question/text_response_comprehension_solution.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+class Course::Assessment::Question::TextResponseComprehensionSolution < ApplicationRecord
+  self.table_name = 'course_assessment_question_text_response_compre_solutions'
+
+  enum solution_type: [:compre_lifted_word, :compre_keyword]
+
+  before_validation :remove_blank_solution,
+                    :strip_whitespace_solution,
+                    :strip_whitespace_solution_lemma
+
+  validate :validate_solution_grade,
+           :validate_solution_grade_is_zero_for_compre_lifted_word
+
+  belongs_to :point, class_name: Course::Assessment::Question::TextResponseComprehensionPoint.name,
+                     inverse_of: :solutions
+
+  def auto_gradable_solution?
+    !solution.empty?
+  end
+
+  def initialize_duplicate(duplicator, other)
+    self.point = duplicator.duplicate(other.point)
+  end
+
+  private
+
+  def comprehension_solution?
+    compre_keyword? || compre_lifted_word?
+  end
+
+  def remove_blank_solution
+    solution.reject!(&:blank?)
+  end
+
+  def strip_whitespace_solution
+    solution.each(&:strip!) unless solution.empty?
```

Comment:
If `solution` is an empty array, then no need to check if empty.  Same for `strip_whitespace_solution_lemma`

Reply:
Will make the changes. Thanks!